contrib/kafka/pkg/dispatcher/dispatcher.go

```diff
@@ -34,15 +34,19 @@ import (
 )
 
 type KafkaDispatcher struct {
-	config     atomic.Value
-	updateLock sync.Mutex
+	// TODO: config doesn't have to be atomic as it is read an updated using updateLock.
+	config           atomic.Value
+	hostToChannelMap atomic.Value
+	updateLock       sync.Mutex
 
 	receiver   *provisioners.MessageReceiver
 	dispatcher *provisioners.MessageDispatcher
 
 	kafkaAsyncProducer sarama.AsyncProducer
-	kafkaConsumers     map[provisioners.ChannelReference]map[subscription]KafkaConsumer
-	kafkaCluster       KafkaCluster
+	// TODO: kafkaConsumer map should probably be atomic as it is updated and read on separate go routines with no syncchronization.
+	// Verify if this is an issue and fix accordignly
+	kafkaConsumers map[provisioners.ChannelReference]map[subscription]KafkaConsumer
```

Comment:
Ideally we can share some code between source (consumer) and channel (consumer), e.g. in `knative/pkg` repo.

I can take that as a task for the "future"

Reply:
Sounds good. Could you please create an issue for it? I'll resolve this comment now.